include/oneapi/dpl/internal/random_impl/discard_block_engine.h

```diff
@@ -129,7 +129,8 @@ class discard_block_engine
 
   private:
     // Static asserts
-    static_assert((0 < _R) && (_R <= _P), "oneapi::dpl::discard_block_engine. Error: unsupported parameters");
+    static_assert((0 < _R) && (_R <= _P) && (_R <= std::numeric_limits<int>::max()),
```

Comment:
I thought we want to limit it to 32 bits. Int doesn't provide this guarantee. Should we change it to std::(u)int32_t?